src/common/pythonInstallation.ts

```diff
@@ -22,10 +22,38 @@ namespace PythonInstallation  {
 			pythonRepository = defaultPythonRepository;
 			pythonRoot = defaultPythonRoot;
 		}
-		if (Uri.parse(pythonRepository).authority !== 'github.com') {
+
+		// Consider third party file system providers
+		try {
+			let pythonRepositoryUri : Uri | undefined = undefined;
+			try {
+				// Uri.parse throws if the URI is invalid
+				pythonRepositoryUri = Uri.parse(pythonRepository);
+			}
+			catch (e) {
+				Tracer.append(`${pythonRepository} is not a valid URI`);
+				throw e;
+			}
+		
+			if (!isGithubUri(pythonRepositoryUri)) {

```

Comment:
Actually we could even do the stat if we are on a GitHub repository. I think it would be good in all cases since even on a GitHub repository users could misconfigure things and we could detect that early on.

Reply:
Is it possible to do so?
My understanding was that at this point, the remotehub may not have loaded the repo into a workspace so `stat` wouldn't be possible without waiting for that.

It is possible with the FSP because the assumption is that the FSP is already available.

Reply:
You can still do the stat. IMO we should look into doing it and see how long it takes. If. a user gives use a github URL it might point to someting non existing as well. In that case it would also make sense to fall back to the default installation.